### ISSUE TITLE
Behaviour change of `apply` when creating evar.

### ISSUE THREAD

**Issue by strub** [math-comp/math-comp#15]
It seems that apply recovered it 1.4 behaviour when creating evars. For an example, see the mail thread in the ssr mailing list entitled "ssreflect 1.5: regression in apply". For instance, in the following example, no more goal is produced for the introduced evar.

```
Section TEST.

Parameter A: Type.
Parameter world: Type.
Definition prop := world -> Prop.
Parameter wle: world-> world -> Prop.

Goal forall (P:prop) (w w':world), wle w w' -> P w ->
   (forall (w : world), P w -> forall (w' : world), wle w w' -> P w') -> 
    P w'.
Proof.
move => P w w' H_wle HP.
apply.
```

**Comment by gares** [comment]
That is intended, the behavior of 1.5 being a backward incompatible change.  Yes, the fix was not quick...

**Comment by strub** [comment]
Just to be sure, the 1.5 -> 1.6 fix is the change you're speaking of ? I.e. that we have two incompatible changes in the picture: the one from 1.4 to 1.5 (when apply started to add goals for evars) and the one from 1.5 to 1.6 when this behaviour has been reverted ?

As a side note, is there a 1.5 -> 1.6 ChangeLog somewhere ?

**Comment by amahboubi** [comment]
There is such a ChangeLog, shipped with the distribution here: 
  https://github.com/math-comp/math-comp/blob/master/etc/ChangeLog.
But it only  mentions the issues with the library. 
The corresponding (and so far more detailed) log for the tactic language is at the end of the manual:
 https://github.com/math-comp/ssr-manual/blob/master/changes.tex
But it looks like we've not pushed content for versions >= 1.5 ...

Thanks for spotting that!

**Comment by gares** [comment]
There is no real changelog for the plugin because nothing really new happened.
Only two fixes, one for apply (that breaks wrt 1.5 and restores 1.4 behavior) and one for elim (that should accept more lemmas I believe).
